Application/EdFi.Ods.Common/Exceptions/EdFiProblemDetailsExceptionBase.cs

```diff
@@ -0,0 +1,131 @@
+  │  └───────────────────┘
+  │  ┌──────────────────────┐
+  ├──┤ ConcurrencyException | 412 Precondition Failed
+  │  └──────────────────────┘
+  │  ┌──────────────────────────────┐
+  ├──┤ InternalServerErrorException | 500 Internal Server Error
+  │  └──────────────────────────────┘
+  │    △
+  │    │  ┌─────────────────────────────┐
+  │    ├──┤ DatabaseConnectionException |
+  │    │  └─────────────────────────────┘
+  │    │  ┌──────────────────────────────┐
+  │    └──┤ SystemConfigurationException |
+  │       └──────────────────────────────┘
+  │         △
+  │         │  ┌──────────────────────────┐
+  │         ├──┤ FeatureDisabledException |
+  │         │  └──────────────────────────┘
+  │         │  ┌─────────────────────────────────────┐
+  │         ├──┤ InvalidSystemConfigurationException |
+  │         │  └─────────────────────────────────────┘
+  │         │    △
+  │         │    │  ┌──────────────────────────┐
+  │         │    └──┤ InvalidApiModelException |
+  │         │       └──────────────────────────┘
+  │         │  ┌────────────────────────────────┐
+  │         └──┤ SecurityConfigurationException |
+  │            └────────────────────────────────┘
+  │  ┌──────────────────────────────────┐
+  ├──┤ ProfileContentTypeUsageException | 400 Bad Request
+  │  └──────────────────────────────────┘
+  │  ┌─────────────────────────────┐
+  ├──┤ ProfileMethodUsageException | 405 Method Not Allowed
+  │  └─────────────────────────────┘
+  │  ┌───────────────────────────────┐
+  ├──┤ SnapshotsAreReadOnlyException | 405 Method Not Allowed
+  │  └───────────────────────────────┘
+  │  ┌──────────────────────┐
+  ├──┤ NotModifiedException | 304 Not Modified
+  │  └──────────────────────┘
+  │
+*/
+
+public abstract class EdFiProblemDetailsExceptionBase : Exception, IEdFiProblemDetails
+{
+    public const string BaseTypePrefix = "urn:ed-fi:api";
+
+    private string _type;
+
+    protected EdFiProblemDetailsExceptionBase(string detail, string message)
+        : base(message)
+    {
+        Detail = detail;
+    }
+
+    protected EdFiProblemDetailsExceptionBase(string detail, string message, Exception innerException)
+        : base(message, innerException)
+    {
+        Detail = detail;
+    }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.Detail" />
+    public string Detail { get; }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.Type" />
+    public string Type
+    {
+        get => _type ??= string.Join(':', GetTypeParts());
+    }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.Title" />
+    public abstract string Title { get; }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.Status" />
+    public abstract int Status { get; }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.CorrelationId" />
+    public string CorrelationId { get; set; }
+
+    /// <inheritdoc cref="IEdFiProblemDetails.ValidationErrors" />
+    Dictionary<string, string[]> IEdFiProblemDetails.ValidationErrors { get; set; }
```

Comment:
`ValidationErrors` and `Errors` are implemented explicitly to prevent them from being logged with the exception. Their contents are intended for the API client, and their end user(s).